packages/flutter/lib/src/animation/animations.dart

```diff
@@ -605,3 +605,21 @@ abstract class CompoundAnimation<T> extends Animation<T>
     }
   }
 }
+
+/// An animation of [double]s that tracks the mean of two other animations.
+///
+/// The [status] of this animation is the status of the `right` animation if it is
+/// moving, and the `left` animation otherwise.
+///
+/// The [value] of this animation is the [double] that represents the mean value
+/// of the values of the `left` and `right` animations.
+class AnimationMean extends CompoundAnimation<double> {
```

Comment:
It was hard to come up with documentation that didn't say the animation was nasty. :-P